src/Framework.UnitTests/SystemSetup_Tests.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using Microsoft.Build.UnitTests;
+using Shouldly;
+using Xunit;
+
+
+namespace Microsoft.Build.Framework
+{
+    public sealed class SystemSetup_Tests
+    {
+        [Fact]
+        public void VerifyLongPaths()
+        {
+            NativeMethodsShared.MaxPath.ShouldBeGreaterThan(10000);
+        }
+
+#if NETCOREAPP
```

Comment:
Why limit this to NETCOREAPP? we support creating symlinks with full-framework msbuild.

Reply:
The only test I know of that requires creating a symlink is core-specific. Also, File/Directory.CreateSymbolicLink are only available on core.

Reply:
Check out https://github.com/dotnet/msbuild/blob/15b127d2445d6b68429f1c31d9ff98fe4e1e728c/src/Tasks.UnitTests/Copy_Tests.cs#L2386-L2409

Reply:
It's entirely nonobvious from that test, but it doesn't create a symlink.

First off, it has a test to make sure you have the right privileges. If you don't, it fails with Assert.True(true, ...)...so it doesn't actually fail.

Then it tries to make a symlink but actually just copies the file. I tried verifying that `(File.GetAttributes(destFile) & FileAttributes.ReparsePoint) != 0`, but that failed. Then I tried breaking shortly after the copy. I modified the file I found at destFile, saved it, and opened sourceFile, and it was unchanged.

I also looked into how to make a symlink on .NET Framework, since I'd agree that it's better to create real symlinks anyway. It sounds like that requires p/invoking into some system call. We try to do that in exactly one place—in the Copy task, as suggested by the test. I looked into why that was failing, and if I understand it right, you need admin privileges to use that particular system call, so that code path isn't normally getting exercised properly.

That is a separate issue. If File.CreateSymbolicLink doesn't require as high privileges as the p/invoke, I think we should consider #ifdeffing that in. For now, since we don't actually have any tests on framework that depend on making symlinks without admin privileges, I'd leave this PR as-is. (Note that when rokonec didn't have privileges to make symlinks, he only saw one test fail.)

Reply:
We talked offline. It's easier to test the functionality of creating symlinks on Core, and it's not fatal to have the check only there since we generally run tests on multiple TFs it should still fail in, for example `build -test`.